### PR TITLE
Automatically create explicit dest-dir for must-gather

### PR DESCRIPTION
**Pull request by tnozicka**
**Description of your changes:**
When using `must-gather` with explicit `--dest-dir=`, this PR allows automatically creating that dir, if it doesn't exist. This helps with the UX so someone iterating on that can only use `rm` but no longer needs the extra `mkdir` command.

### CODE REVIEW

File: pkg/cmd/operator/gatherbase.go [hunk 1]
```diff
@@ -107,13 +103,24 @@ func (o *GatherBaseOptions) Complete() error {
...
 
+	ignoreDestDirExists := false
 	if len(o.DestDir) == 0 {
 		o.DestDir = fmt.Sprintf("%s-%s", o.GathererName, utilrand.String(12))
-		err := os.Mkdir(o.DestDir, 0770)
-		if err != nil {
-			return fmt.Errorf("can't create destination directory %q: %w", o.DestDir, err)
-		}
+	} else {
+		// We have already made sure that the dir doesn't exist or is empty in validation.
```
Comment (rzetelskik) [inline pkg/cmd/operator/gatherbase.go:110]: Wouldn't the logic be simpler if you moved DestDir creation before verification?

**Review by rzetelskik — COMMENTED** [review]
/lgtm

File: pkg/cmd/operator/gatherbase.go (hunk 1 above)
Comment (rzetelskik) [reply pkg/cmd/operator/gatherbase.go:110]: /hold

Comment (tnozicka) [reply pkg/cmd/operator/gatherbase.go:110]: I don't think it should create dest dir when there are invalid arguments. The point of verification as a first step is to make sure the inputs are correct before taking any action.

Comment (rzetelskik) [reply pkg/cmd/operator/gatherbase.go:110]: Fair enough. In that case I'd personally move the destDir creation out of the Validate func though.

Comment (tnozicka) [reply pkg/cmd/operator/gatherbase.go:110]: it's inside "Complete()" function...

Comment (rzetelskik) [reply pkg/cmd/operator/gatherbase.go:110]: ah, nvm, didn't notice the lines skipped by github

**Review by rzetelskik — APPROVED** [review]
/lgtm

**Pull request merged**